src/qibolab/couplers.py

```diff
@@ -25,14 +25,28 @@ class Coupler:
     native_pulse: CouplerNatives = field(default_factory=CouplerNatives)
     "For now this only contains the calibrated pulse to activate the coupler."
 
-    flux: Optional[Channel] = None
+    _flux: Optional[Channel] = None
     "flux (:class:`qibolab.platforms.utils.Channel`): Channel used to send flux pulses to the qubit."
 
     # TODO: With topology or conectivity
     # qubits: Optional[Dict[QubitId, Qubit]] = field(default_factory=dict)
     qubits: Dict = field(default_factory=dict)
     "Qubits the coupler acts on"
 
+    def __post_init__(self):
+        if self.flux is not None and self.sweetspot != 0:
+            self.flux.offset = self.sweetspot
+
+    @property
+    def flux(self):
+        return self._flux
+
+    @flux.setter
+    def flux(self, channel):
+        if self.sweetspot != 0:
+            channel.offset = self.sweetspot
```

Comment:
Can't you just set in any case, even when `self.sweetspot == 0`?

Would it be wrong for some reason?